src/server_manager/web_app/testing/models.ts

```diff
@@ -0,0 +1,222 @@
+// Copyright 2021 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as cloud from '../../model/cloud';
+import * as digitalocean from '../../model/digitalocean';
+import * as gcp from '../../model/gcp';
+import * as server from '../../model/server';
+
+export class FakeDigitalOceanAccount implements digitalocean.Account {
+  private servers: server.ManagedServer[] = [];
+
+  constructor(private accessToken = 'fake-access-token') {}
+
+  async getName(): Promise<string> {
+    return 'fake-digitalocean-account-name';
+  }
+  async getStatus(): Promise<digitalocean.Status> {
+    return digitalocean.Status.ACTIVE;
+  }
+  listServers() {
+    return Promise.resolve(this.servers);
+  }
+  getRegionMap() {
+    return Promise.resolve({'fake': ['fake1', 'fake2']});
+  }
+  createServer(id = Math.random().toString()) {
```

Comment:
I think this won't work as intended. You will use the same random id on every call.
Better to always pass, or set it if absent in the function body.